### PR TITLE
added Redux to the project, created store dir and all the general stu…

### PR DESCRIPTION
**Pull request by ritasumtsova**
1. Added Redux, React-Redux, Redux-Thunk and @types stuff to the project;
2. Created store dir with the store.ts, rootReducer.ts + departmentsActionCreators/departmentsReducer/departmentsActionTypes.ts
3. Added a store folder to the types dir to describe all the needed types/interfaces
4. Changed Departments.tsx to use redux for the getDepartments() API call.

### CODE REVIEW

File: src/components/DepartmentsList/DepartmentsList.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ interface DepartmentsListProps {
 
 const DepartmnetsList: React.FC<DepartmentsListProps> = ({ departments }) => {
   if (departments.length === 0) {
-    return <NotFound />;
+    return <NotFound errMsg=""/>;
```
Comment (didevgen) [inline src/components/DepartmentsList/DepartmentsList.tsx:17]: 1. what for? :)
2. chng t errorMessage  pls

File: src/components/pages/Departments/Departments.tsx [hunk 2]
```diff
@@ -1,40 +1,27 @@
...
-
-    try {
-      const res = await DepartmentsAPI.getDepartments();
-
-      setDepartments(res.data.data);
-    } catch (err) {
-      console.error(err);
-    }
+  const departments = useSelector((state: rootState) => state.departments.departments?.data);
+  const errMsg = useSelector((state: rootState) => state.departments.errorMsg);
```
Comment (didevgen) [inline src/components/pages/Departments/Departments.tsx:12]: selectors should come from different file

File: src/store/actionCreators/departmentsActionCreators.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+};
+
+export const fetchDepartmentsFailure = (errorMsg: string): ActionType => {
+  return {
+    type: departmentsActionTypes.FETCH_DEPARTMENTS_FAILURE,
+    payload: errorMsg
+  };
+};
+
+export const fetchDepartmentsThunk = (): AppThunk => {
```
Comment (didevgen) [inline src/store/actionCreators/departmentsActionCreators.ts:29]: put into different file

File: src/components/DepartmentsList/DepartmentsList.tsx (hunk 1 above)
Comment (ritasumtsova) [reply src/components/DepartmentsList/DepartmentsList.tsx:17]: there's no need actually I did just to render error message it the NotFound component for now

File: src/components/pages/Departments/Departments.tsx (hunk 2 above)
Comment (ritasumtsova) [reply src/components/pages/Departments/Departments.tsx:12]: OK thanks

**Review by didevgen — APPROVED** [review]

File: src/components/DepartmentsList/DepartmentsList.tsx (hunk 1 above)
Comment (didevgen) [reply src/components/DepartmentsList/DepartmentsList.tsx:17]: you can make errorMessage optional param

File: src/store/actionCreators/departmentsActionCreators.ts (hunk 3 above)
Comment (ritasumtsova) [reply src/store/actionCreators/departmentsActionCreators.ts:29]: got it

**Pull request merged**